woocommerce/woocommerce-ios
# Order Creation: Make Payment section always visible on New Order screen

Closes: #6262

## Description

During Order Creation, the Payment section on the New Order screen is now always visible, instead of appearing after products are added to the order. This way, shipping and fees can be added to an order without products.

Note: This change is related to the Shipping & Fees features that are part of Milestone 2, but I did not put this change behind a feature flag so it will also affect the current experimental feature behavior (with the Milestone 1 features). I chose to do this because it simplifies the required changes, and because we're almost ready to release at least Shipping as part of the experimental Order Creation feature. Let me know if you think we should still put this change behind a feature flag, anyway.

## Changes

Removes the check for `shouldShowPaymentSection` (which checks for items in the order) to display `OrderPaymentSection` in the `NewOrder` view.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Confirm the Payment section is immediately visible on the New Order screen, with a "Product total" line showing a $0 value.
5. Add products, shipping, and/or fees to the order and confirm the Payment section updates as expected.

You can repeat these steps with the `.orderCreation` feature flag disabled (to test the experience with M1 features only) or with the `.orderCreationRemoteSynchronizer` feature flag enabled (to test it with the WIP remote sync feature).

## Screenshots

Milestone 1 features|Milestone 2 features
-|-
![Simulator Screen Shot - iPhone 13 Pro - 2022-02-22 at 15 55 38](https://user-images.githubusercontent.com/8658164/155171097-3dc74d65-8faa-4167-97cc-5f9ac19ce451.png)|![Simulator Screen Shot - iPhone 13 Pro - 2022-02-22 at 15 27 11](https://user-images.githubusercontent.com/8658164/155171112-837a69b2-17bd-4f34-9e7c-49c7207b6a61.png)


## Submitter Checklist

Update release notes:
